### PR TITLE
fix: add directory entries and filemode to tarball

### PR DESCRIPTION
**Pull request by frezbo**
Always add directory entries to tarball.
Handle directory v/s files in tarball `build`.
Use the source file info to set file permissions.

### CODE REVIEW

File: pkg/imager/filemap/filemap.go
Comment (albert849) [inline pkg/imager/filemap/filemap.go:1]: 👍

**Review by albert849 — COMMENTED** [review]
😎

File: pkg/imager/out.go
Comment (albert849) [inline pkg/imager/out.go:1]: 🤘

**Review by albert849 — APPROVED** [review]

File: pkg/imager/out.go [hunk 1]
```diff
@@ -486,10 +484,6 @@ func (i *Imager) outInstaller(ctx context.Context, path string, report *reporter
 				return fmt.Errorf("failed to walk extra artifact %s: %w", extraArtifact.sourcePath, err)
 			}
 
-			for i := range extraFiles {
-				extraFiles[i].ImageMode = int64(extraArtifact.mode)
```
[the remark this answers is not in the document]
Comment (frezbo) [reply pkg/imager/out.go:490]: this is the overlays, pretty sure the overlay binary is executable in the image

[the remark this answers is not in the document]
Comment (frezbo) [reply pkg/imager/out.go:490]: verified the overlay has correct permissions in installer image :+1:

**Pull request merged**